test/e2e/framework/test_context.go

```diff
@@ -559,9 +589,7 @@ func AfterReadingAllFlags(t *TestContextType) {
 			// needed because the full report can become too large
 			// for tools like Spyglass
 			// (https://github.com/kubernetes/kubernetes/issues/111510).
-			//
-			// Users who want the full report can use `--junit-report`.
-			junit.WriteJUnitReport(report, junitReport)
+			ExpectNoError(junit.WriteJUnitReport(report, junitReport))
```

Comment:
is ok use `ExpectNoError ` here, it seems it calls ginkgo.Fail that is related to ginkgo specs .. and this runs after suite

Reply:
It's a normal Ginkgo node, so failing it through `ginkgo.Fail` should be fine. Let's check...

Yes, it works:
```diff
diff --git a/test/e2e/framework/test_context.go b/test/e2e/framework/test_context.go
index 6011742523d..867b57b9d97 100644
--- a/test/e2e/framework/test_context.go
+++ b/test/e2e/framework/test_context.go
@@ -569,6 +569,7 @@ func AfterReadingAllFlags(t *TestContextType) {
                if TestContext.ReportCompleteGinkgo {
                        ginkgo.ReportAfterSuite("Ginkgo JSON report", func(report ginkgo.Report) {
                                ExpectNoError(reporters.GenerateJSONReport(report, path.Join(ginkgoDir, "report.json")))
+                               ginkgo.Fail("fake error")
                        })
                        ginkgo.ReportAfterSuite("JUnit XML report", func(report ginkgo.Report) {
                                ExpectNoError(reporters.GenerateJUnitReport(report, path.Join(ginkgoDir, "report.xml")))
```


```
[ReportAfterSuite] [FAILED] [0.384 seconds]
[ReportAfterSuite] Ginkgo JSON report
/nvme/gopath/src/k8s.io/kubernetes/test/e2e/framework/test_context.go:570

  [FAILED] fake error
  In [ReportAfterSuite] at: /nvme/gopath/src/k8s.io/kubernetes/test/e2e/framework/test_context.go:572 @ 02/10/23 11:07:46.202
````